From: Ubuntu Desktop Team <ubuntu-desktop@lists.ubuntu.com>
Date: Thu, 26 Jul 2018 16:19:06 +0100
Subject: media-keys: correct logout action

let the logout key do what it's supposed to do

Bug-Ubuntu: https://launchpad.net/bugs/961501
Bug-GNOME: https://bugzilla.gnome.org/show_bug.cgi?id=671979
---
 plugins/media-keys/gsd-media-keys-manager.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/plugins/media-keys/gsd-media-keys-manager.c b/plugins/media-keys/gsd-media-keys-manager.c
index d923978..9ed0637 100644
--- a/plugins/media-keys/gsd-media-keys-manager.c
+++ b/plugins/media-keys/gsd-media-keys-manager.c
@@ -1168,6 +1168,12 @@ gnome_session_shutdown (GsdMediaKeysManager *manager)
         g_object_unref (proxy);
 }
 
+static void
+do_logout_action (GsdMediaKeysManager *manager)
+{
+        execute (manager, "gnome-session-quit --logout", FALSE);
+}
+
 static void
 do_eject_action_cb (GDrive              *drive,
                     GAsyncResult        *res,
@@ -2685,7 +2691,7 @@ do_action (GsdMediaKeysManager *manager,
                                  SOUND_ACTION_FLAG_IS_OUTPUT | SOUND_ACTION_FLAG_IS_PRECISE);
                 break;
         case LOGOUT_KEY:
-                gnome_session_shutdown (manager);
+                do_logout_action (manager);
                 break;
         case EJECT_KEY:
                 do_eject_action (manager);
